python/pyiceberg/io/pyarrow.py

```diff
@@ -749,7 +869,16 @@ def _task_to_table(
             schema_raw = metadata.get(ICEBERG_SCHEMA)
         # TODO: if field_ids are not present, Name Mapping should be implemented to look them up in the table schema,
         #  see https://github.com/apache/iceberg/issues/7451
-        file_schema = Schema.parse_raw(schema_raw) if schema_raw is not None else pyarrow_to_schema(physical_schema)
+        file_schema = (
+            Schema.parse_raw(schema_raw)
+            if schema_raw is not None
+            else pyarrow_to_schema(
+                physical_schema,
+                projected_schema,
```

Comment:
I think we may want full table schema here. `pyarrow_to_schema` supposed to simply convert the physical schema to the iceberg schema without handling column pruning. However, `projected_schema` only contains selected columns in a table scan. If we use it during the conversion, we will have to ignore unselected columns, which I think is unnecessary and tricky to implement. (also inconsistent with the behavior when field_id is present)

Reply:
It doesn't have much to do with the pruning. What we need to acheive with `ignore_unprojected_fields` here is to simply ignore redundant columns in the actual data, and the purpose of pruning is to take away the fields that are already *known* according to the catalog.  Those are similar, but have different semantics.